source/fundamentals/crud/write-operations/insert.txt

```diff
@@ -4,9 +4,194 @@ Insert a Document
+the new document.
+
+Example
+~~~~~~~
+
+The following example creates and inserts a document into the
+``favorite_books`` collection using the
+``InsertOne()`` method:
+
+.. code-block:: go
+
+   coll := client.Database("myDB").Collection("favorite_books")
+   doc := bson.D{{"title", "Invisible Cities"}, {"author", "Italo Calvino"}, {"year_published", 1974}}
+
+   result, err := coll.InsertOne(context.TODO(), doc)
+
+   fmt.Printf("Inserted document with _id: %v\n", result.InsertedID)
+
+The output should look like this:
+
+.. code-block:: none
+   :copyable: false
+
+   Inserted document with _id: ObjectID("6112c7cacf7fb532a9c0077f")
+
+For more information on the types and functions mentioned in this
+section, see the following resources:
+
+- `InsertOne() <https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo@v1.7.0#Collection.InsertOne>`__  API documentation
+- `InsertOneResult <https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo@v1.7.0#InsertOneResult>`__ API documentation
+- Server Manual Entry on :manual:`insertOne() </reference/method/db.collection.insertOne/>`
+- Runnable :doc:`Insert a Document Usage Example </usage-examples/insertOne>`
+
+Insert Multiple Documents
+-------------------------
+
+Use the ``InsertMany()`` method to insert multiple documents into a
+collection.
+
+Upon successful insertion, the ``InsertMany()`` method returns an ``InsertManyResult``
+instance that contains the ``_id`` fields of the inserted documents.
+This method inserts documents in the order specified unless an exception
+occurs.
+
+For example, assume you want to insert the following documents:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id": 1, "country": "Tanzania" }
+   { "_id": 2, "country": "Lithuania" }
+   { "_id": 1, "country": "Vietnam" }
+   { "_id": 3, "country": "Argentina" }
+
+If you attempt to insert these documents, a ``WriteError`` occurs at the
+third document because of the repeated ``_id`` value, but the documents
+before the error-producing document still get inserted into your collection.
+
+.. note::
+
+   You can get an acknowledgement of successful document insertion even
+   if a WriteError occurs:
+
+   .. code-block:: go
+      
+      docs := []interface{}{
+	      bson.D{{"_id", 1}, {"country", "Tanzania"}},
+	      bson.D{{"_id", 2}, {"country", "Lithuania"}},
+	      bson.D{{"_id", 1}, {"country", "Vietnam"}},
+	      bson.D{{"_id", 3}, {"country", "Argentina"}},
+	  }
+
+      result, err := coll.InsertMany(context.TODO(), docs)
+      list_ids := result.InsertedIDs
+      if err != nil {
+	      fmt.Printf("A bulk write error occurred, but %v documents were still inserted.\n", len(list_ids))
+      }
+
+      for _, id := range list_ids {
+	      fmt.Printf("Inserted document with _id: %v\n", id)
+      }
+   
+   The output should look like this:
+
+   .. code-block:: none
+      :copyable: false
+      
+      A bulk write error occurred, but 2 documents were still inserted.
```

Comment:
Nit: The wording would be incorrect if there was only one document -- maybe change the second clause to "but the following documents were inserted:"  and print the array/interface of the _id?